src/main/java/com/artipie/asto/fs/FileStorage.java

```diff
@@ -145,6 +145,15 @@ public FileStorage(final Path path, final FileSystem fls) {
             .toCompletableFuture();
     }
 
+    @Override
+    public CompletableFuture<Void> delete(final Key key) {
+        return CompletableFuture.supplyAsync(
+            () -> new RxFile(this.path(key), this.fls)
```

Comment:
@olegmoz can we use just `return new RxFile(this.path(key), this.fls).delete().to(CompletableInterop.await())`? Right now `delete` method will spawn new thread (or use tread pull) for creating new instance of `RxFile`.

Reply:
@g4s8 fixed that